### ISSUE TITLE
guest vms on covered hypervisors

### ISSUE THREAD

**Issue by setharnold** [canonical/ubuntu.com#12592]
Hello, the text on the Pro pricing page includes:

> Server with unlimited VMs*

Does this mean that a single Pro license on the hypervisor can be used on all the guests on the machine? Or does this simply not limit the number of guests? I didn't spot any clear guidance if every VM Guest needs its own Pro license or not.

Thanks

---
*Reported from: https://ubuntu.com/pricing/pro*

**Comment by cpaelzer** [comment]
TL;DR plan of action (IMHO, just a suggestion):

- step 1: @lechsandecki to clarify the wording on [ubuntu-pro-description][2]
- step 2: @lechsandecki maybe add a link to [ubuntu-pro-description][2] at the "*" explanation for unlimited VMs in [pro-pricing][6]
- step 3: @lechsandecki to extend the [FAQ][4] with this
- step 4: @s-makin to carry anything that comes out of this also to the docs of the pro client
- step 5: @cpaelzer to carry any implications to code or help-texts of the client to the [Pro client][5]
- step 6: ensure any similar questions that have piled up until then will be answered pointing to the updated descriptions

--- details ---

Thanks Seth, that is indeed worth to clarify!
It did not come up as a frequently asked question before, but just after @lechsandecki posted the [FAQ][4] people seem to have overcome the initial confusion and this seems to be one of the follow up questions that start to line up now.

There also is a similar [reddit post][1]:

> Can all the vms be updated for 10 years without upgrading?

And we also have another incarnation of the same question in [askubuntu][3]:

> I'm a little bit at a loss understanding how Ubuntu Pro license works for Vms. It says "all the VMs running on these hosts", then listing a lot of hosts (from qemu/KVM to EXSi to Hyper-V to others), and "every node in the cluster must be subscribed".
> 
> How can I subscribe a EXSi node/host? Is there like a dedicated client? And what about Proxmox (qemu/KVM based)? Should I just subscribe the VMs and the hosts gets automatically "subscibed"? How does it work actually?

Given the frequency of occurrence of these questions we should really clarify the full wording (@lechsandecki that would be you I think).
This is then probably going to happen on [ubuntu-pro-description][2] first and then seems worth to also be added to the [FAQ][4].

Following to that we'd check if that has any implications to code, help texts or documentation (@s-makin to track this for ensuring any outcome is in our docs as well eventually) which in turn we would have a look for that on our side in the [Pro client][5].

What do others think about this plan of action?

[1]: https://www.reddit.com/r/Ubuntu/comments/119l57y/does_ubuntu_pro_include_support_for_each_vm_on_a/
[2]: https://ubuntu.com/legal/ubuntu-pro-description
[3]: https://askubuntu.com/questions/1456335/ubuntu-pro-license-for-vms
[4]: https://discourse.ubuntu.com/t/ubuntu-pro-faq/34042
[5]: https://github.com/canonical/ubuntu-advantage-client
[6]: https://ubuntu.com/pricing/pro

**Comment by cpaelzer** [comment]
FYI - I answered the askubuntu question with the state of today to not let it dangle unanswered for too long.
We might revise my quick answer later once we have completed this clarification here.

@s-makin for the docs of the pro-client what would you think of something like an "how to attach ..." for all variations that we know and support. That should IMHO include an overview that maps the text (which is correct, but seems misunderstandable) into a drawing that would help understanding the text. Here a quick sketch
**(please no one take this as answer, I'm proposing ideas not yet more).**

![ubuntu pro vm support clarification (2)](https://user-images.githubusercontent.com/14993822/221117140-993525f7-87fd-4112-a69d-7cb4c2320c4c.png)

**Comment by urbaman** [comment]
Hi, thank you very much, I'm the one behind the askubuntu incarnation of the request.

Another thing that's probably more technical but useful in pre-sale: if I say, have a three-node proxmox cluster (debian KVM/qemu based, so hopefully covered), and buy 3x license (one per node), will I have a single token/license or three for the cluster?

Thing is, the VMs will probably travel between nodes during their life (for high availability and/or node management situatuions). So, will I have to switch the moved VMs between tokens/licenses or will they just be handled without intervention?

Thank you again

**Comment by cpaelzer** [comment]
> will I have a single token/license or three for the cluster?

If you buy one subscription with 3 nodes you will have one token for your whole subscription, no need to change it on migration.

**Comment by lechsandecki** [comment]
thanks for raising that @setharnold, and @cpaelzer I love the clarification you proposed

1. Both the pricing page and the ubuntu pro description are quite clear in terms of the scope. Specifically:

"* Any of: KVM | Qemu | Boch, VMWare ESXi, LXD | LXC, Xen, Hyper-V (WSL, Multipass), VirtualBox, z/VM, Docker. All Nodes in the cluster have to be subscribed to the service in order to benefit from the unlimited VM support"
and
"A subscription attached to a physical host running Ubuntu or a Covered Hypervisor. If all physical hosts in the Environment are attached, then Ubuntu Pro subscription also covers all Ubuntu guests on those hosts"

2. I think that the only part that needs an explanation is the process to follow:
I am happy to address that in the FAQ by adding "how can I benefit from the unlimited VM coverage?" and using what @cpaelzer has suggested above.

WDYT?
